docs/guide/provider-migration.md

```diff
@@ -1,3 +1,147 @@
+```
+
+### Handling the Removal of `isEnabled()` and `isApproved()`
+
+Before the new provider API shipped, we added
+[`_metamask.isEnabled`](./ethereum-provider.html#ethereum-metamask-isenabled-to-be-removed) and
+[`_metamask.isApproved`](./ethereum-provider.html#ethereum-metamask-isapproved-to-be-removed)
+to enable web3 sites to check if they have [access to the user's accounts](./rpc-api.html#eth-requestaccounts).
+`isEnabled` and `isApproved` are identical, except that `isApproved` is `async`.
+These methods were arguably never that useful, but they are completely redundant since the introduction of MetaMask's [permission system](./rpc-api.html#permissions).
+
+We recommend that you check for account access in the following ways:
+
+1. You can call the [`wallet_getPermissions` RPC method](./rpc-api.html#wallet-getpermissions) and check for the `eth_accounts` permission.
+
+2. You can call the `eth_accounts` RPC method and the [`ethereum._metamask.isUnlocked()` function](./ethereum-provider.html#ethereum-metamask-isunlocked).
+
+   - MetaMask has to be unlocked before you can access the user's accounts.
+     If the array returned by `eth_accounts` is empty, check if MetaMask is locked using `isUnlocked()`.
+
+   - If MetaMask is unlocked and you still aren't receiving any accounts, it's time to request accounts using the [`eth_requestAccounts` RPC method](./rpc-api.html#eth-requestaccounts).
+
+## Replacing `window.web3`
+
+For historical reasons, MetaMask injects [`web3@0.20.7`](https://github.com/ethereum/web3.js/tree/0.20.7) into all web pages.
+This version of `web3` is effectively deprecated, and new features are no longer added by the [web3.js](https://github.com/ethereum/web3.js/) team.
+
+If your website relies on our `window.web3` object, your Ethereum-related functionality will break when we stop injecting `window.web3`.
+Continue reading to learn more about the migration options.
+
+::: tip
+Regardless of how you choose to migrate, you will probably want to read the `web3@0.20.7` documentation, which you can find [here](https://github.com/ethereum/web3.js/blob/0.20.7/DOCUMENTATION.md).
+:::
+
+### Using `window.ethereum` Directly
+
+For many web3 sites, the API provided by `window.ethereum` is more than sufficient.
+Much of the `web3` API simply maps to RPC methods, all of which can be requested using [`ethereum.request()`](./ethereum-provider.html#ethereum-request-args).
+For example, here are a couple of actions performed using first `window.web3`, and then their equivalents using `window.ethereum`.
+
+```javascript
+/**
+ * Getting Accounts
+ */
+
+// window.web3
+const accounts = web3.eth.accounts;
+
+// window.ethereum
+const accounts = await ethereum.request({ method: 'eth_accounts' });
+
+/**
+ * Sending a Transaction
+ */
+
+// window.web3
+web3.eth.sendTransaction(
+  {
+    to: '0x...',
+    amount: '0x...',
+    // And so on...
+  },
+  (result) => {
+    // Handle the result
+  }
+);
+
+// window.ethereum
+ethereum
```

Comment:
Hm, we actually only use `try/catch` in 2 places, `.then` in 17 places, and `.catch` in 14 places. 😑 

I'll us `try/catch` here, bringing the total to 3, but, for our future reference.

Reply:
#139

Reply:
37672b2

Reply:
`async/await` vs. `.then/.catch` seems like the more relevant distinction. Most of the examples don't handle errors, which is fine for `async/await` because they just bubble up. I figured it was worth showing the error handling here explicitly for symmetry with the callback style example though.